### ISSUE TITLE
Formalize O2 config profiles

### ISSUE THREAD

**Issue by ArtemSokolov** [labsyspharm/mcmicro#73]
We discussed having `O2small`, `O2medium` and `O2large`. To begin, this issue will document all successful runs with sizes of the input data and cluster options used. Once enough data is collected, formal profiles can be created that best cover the various dataset size options.

Ajit successfully ran the pipeline on a TMA with 257 cores using the following cluster options:
```
illumination {clusterOptions = '-t 0-10:00 --mem=16000'}
dearray {clusterOptions = '-t 0-12:00 -c 15 --mem=128000'}
ashlar {clusterOptions = '-t 0-10:00 --mem=64000'}
unmicst {clusterOptions = '-t 0-10:00 --mem=128000'}
s3seg {clusterOptions = '-t 0-10:00 --mem=128000'}
quantification {clusterOptions = '-t 0-10:00 --mem=128000'}
```

**Comment by ArtemSokolov** [comment]
Janae successfully de-arrayed a TMA with 123 cores using:
```
dearray {clusterOptions = '-t 1:00:00 -c 10 --mem=64000'}
```

**Comment by ajitjohnson** [comment]
Ajit successfully ran the pipeline on a TMA with 257 cores using the following cluster options:

```
illumination {clusterOptions = '-t 0-10:00 --mem=16000'} #notused
dearray {clusterOptions = '-t 0-04:00 -c 15 --mem=128000'}
ashlar {clusterOptions = '-t 0-10:00 --mem=64000'} #notused
unmicst {clusterOptions = '-t 0-01:00 --mem=128000'}
s3seg {clusterOptions = '-t 0-01:00 --mem=64000'}
quantification {clusterOptions = '-t 0-02:00 --mem=64000'}
```

**Comment by clarenceyapp** [comment]
I think we should have profiles for 1) large tissue, 2) small tissue, and 3) TMA. If the sample is a TMA, then ashlar, illumination and coreograph should almost always use the large tissue profile settings but unmicst, s3seg, and quantification should be from the small tissue profile since they would have been split into tiny images already.

**Comment by DenisSch** [comment]
I receive out of memory issues on large TNP samples from Jerry. I will try for unmicst
```
--mem=256000
```

**Comment by ArtemSokolov** [comment]
Latest data point for Ajit's 257-core TMA:

Ajit "ran the pipeline with 1G memory usage and 30 minutes as compute time- some of cores passed unmist, some failed with time-out error."

**Comment by ArtemSokolov** [comment]
Janae successfully processed a 123-core TMA with the current O2large profile:

```
withName:illumination   {clusterOptions = '-t 1:00:00 --mem=16000'}         # Not used
withName:dearray        {clusterOptions = '-t 1:00:00 -c 10 --mem=64000'}
withName:ashlar         {clusterOptions = '-t 6:00:00 --mem=64000'}         # Not used
withName:unmicst        {clusterOptions = '-t 12:00:00 --mem=128000'}
withName:s3seg          {clusterOptions = '-t 12:00:00 --mem=128000'}
withName:quantification {clusterOptions = '-t 12:00:00 --mem=128000'}
```

In line with earlier comments, unmicst through quantification requirements can be substantially lower for TMAs.

**Comment by ajitjohnson** [comment]
I processed a 258-core TMA with the following profile

```
illumination   {clusterOptions = '-t 1:00:00 --mem=16000'} # Not used
dearray        {clusterOptions = '-t 6:00:00 -c 15 --mem=128000'}
ashlar         {clusterOptions = '-t 6:00:00 --mem=64000'} # Not used
unmicst        {clusterOptions = '-t 04:00:00 --mem=64000'}
s3seg          {clusterOptions = '-t 04:00:00 --mem=64000'}
quantification {clusterOptions = '-t 04:00:00 --mem=64000'}
```

**Comment by ArtemSokolov** [comment]
Addressed by #106
